### ISSUE TITLE
[bugreport] ZIP file missing dependancies (ie provider dll's)

### ISSUE THREAD

**Issue by morphje** [ebekker/ACMESharp#189]
There appears to be difference between the nupkg file and the zip file.
The following 7 files are missing from the zip archive:
- microsoft.web.administration.dll
- ACMESharp.providers.*
- ACMESharp.POSH-test*

These files prevent the IIS handler to be called from the Complete-ACMEChallenge method, resulting in the error "The given key was not present in the dictionary"

workaround: download the nupkg, unzip, copy files from to your zip archive.

**Comment by yamlfd** [comment]
Where can I download all these files and where do I need to place them?

**Comment by ebekker** [comment]
Did you pull the .zip file from the releases page (https://github.com/ebekker/ACMESharp/releases)?  I assume you're referring to the `ACME-posh.zip` file?

**Comment by yamlfd** [comment]
I have installed from PSGallery as recommended ant it is completly broken. That is such a serious bug... can you not release a fixed version, please?

**Comment by ebekker** [comment]
If you have installed from PowerShellGallery, then all of the required components should be in there, the current release (0.8.1) already includes all the providers that were available at that time.

Whey are you looking for the POSH-test* assemblies?  That only includes test code?

**Comment by ebekker** [comment]
If you pulled from PSGallery, you should not be messing with a .zip file?  You said that's what's missing some files?  I'm confused as to what the actual problem is here?

**Comment by yamlfd** [comment]
I have installed from PSGallery and I always receive "The given key was not present in the dictionary" when running Complete-ACMEChallenge. Based on above reporter I seem to be affected by the issue reported here. It just shows me the recommended install on a brand new system simply does not work out of the box

I just asked how to workaround until a working release exists.

I have upgraded to powershell 5.0 on windows 2012 core. .net is installed. After this I installed the ps module from psgallery.

**Comment by morphje** [comment]
Yes i can conform that I pulled the zip file from the releases. To be exact: https://github.com/ebekker/ACMESharp/releases/download/v0.8.1.0-EA/ACME-posh.zip

I'm only reporting the differences between the zip archive and the nupkg. I haven't compared the psgallery package. I figured that the test files are not for production, but it's a difference between the packages and the zip archive doesn't work without those providers. (well at least the IIS one ;) )

@yamlfd the workaround is easy. On the releases page you will find the nupkg file. Just unzip it, locate the files and copy to wherever your psgallery package has been installed (I haven't worked much with psgallery, so i can't tell you where it's located exactly)

**Comment by morphje** [comment]
@ebekker i'm using the zipfile, because in my current workflow it's easier to copy files and execute my script. I'm using Ansible to bootstrap my servers. Ansible doesn't have an native PSGallery module, so i'm back to copying files. That's why I'm using the ZIP archive.

By the way: i can confirm that my powershell script currently works as intented and i'm happy you created the ACMESharp modules. Please keep up the good work.
